### PR TITLE
MGDOBR-1128: add stage cron job

### PR DESCRIPTION
**Pull request by winklerm**
[MGDOBR-1128](https://issues.redhat.com/browse/MGDOBR-1128)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by winklerm** [comment]
Seems that sending an email from a cronjob is not a built-in feature, will need to check further.

**Review by msmagnanijr — APPROVED** [review]
lgtm!  Let us know if it worked... and then we'll ask someone else to review it and we can do the merge :D

**Comment by winklerm** [comment]
Thanks, will test it tomorrow and let you know!

File: app-interface/cronjobs/rhose-stage-prolong.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                          echo "Prolonging cluster $CLUSTER_NAME with ID $CLUSTER_ID"
+                          printf '{\n\t"expiration_timestamp": "%s"\n}\n' "$(date --iso-8601=seconds -d '+7 days')" | ocm patch /api/clusters_mgmt/v1/clusters/${CLUSTER_ID}
+                        done
+                    env:
+                    - name: CLUSTER_NAMES
+                      value: ${CLUSTER_NAMES}
+                    - name: STAGE_TOKEN
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${STAGE_CRON_JOB_SECRET_NAME}
```
Comment (danielezonca) [inline app-interface/cronjobs/rhose-stage-prolong.yml:55]: As far as I see this assume that we have a single stage cluster. It should be ok and we can always create another similar job if necessary.

**Review by danielezonca — APPROVED** [review]

File: app-interface/cronjobs/rhose-stage-prolong.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+---
+apiVersion: template.openshift.io/v1
```
Comment (danielezonca) [inline app-interface/cronjobs/rhose-stage-prolong.yml:2]: Is this job deployed also in prod? We only need it in stage. I don't understand if this is going to apply everywhere

**Review by robpblake — APPROVED** [review]

**Comment by winklerm** [comment]
Tested on the stage cluster (created manually without app-interface involvement yet).

**Pull request merged**